### ISSUE TITLE
Printing objects containing cint via strformat gives deprecation warning related to `addInt`

### ISSUE THREAD

**Issue by kaushalmodi** [nim-lang/Nim#11314]
Printing objects containing cint via strformat gives deprecation warning related to `addInt`

### Example

```nim
import strformat

type
  ObjTypeWithInt = object
    i: cint

let
  o1 = ObjTypeWithInt(i: 101.cint)
echo &"o1 = {o1}"
```

### Current Output
The output is correct but it prints this warning:

```
ocal/apps/6/nim/devel/lib/pure/strformat.nim(514, 23) template/generic instantiation of `$` from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/system/dollars.nim(90, 15) template/generic instantiation of `addQuoted` from here
/home/kmodi/usr_local/apps/6/nim/devel/lib/system.nim(4210, 7) Warning: Deprecated since v0.20, use 'addInt'; add is deprecated [Deprecated]
```

### Expected Output

No warnings should appear.

### Additional Information

This is related to https://github.com/nim-lang/Nim/pull/11292 (/cc @narimiran ).

```
$ nim -v
Nim Compiler Version 0.19.9 [Linux: amd64]
Compiled at 2019-05-23
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: 44cc5f6360c7ccc96c296948b2524bd2cdebf1f0
active boot switches: -d:release
```

**Comment by narimiran** [comment]
Argh, line in `addQuoted` should be changed to `elif T is (int or cint) and not defined(JS)` I guess. (Currently is only `int`)

**Comment by kaushalmodi** [comment]
and then similarly for float/cfloat? (what about other int/cint/float/cfloat variations: signed/unsigned 8/16/32/64 bits?)

**Comment by kaushalmodi** [comment]
May be do something like:

```diff
diff --git a/lib/system.nim b/lib/system.nim
index 421973cce..ae243dcc7 100644
--- a/lib/system.nim
+++ b/lib/system.nim
@@ -4202,9 +4202,9 @@ proc addQuoted*[T](s: var string, x: T) =
     s.addEscapedChar(x)
     s.add("'")
   # prevent temporary string allocation
-  elif T is int and not defined(JS):
+  elif T is SomeInteger and not defined(JS):
     s.addInt(x)
-  elif T is float and not defined(JS):
+  elif T is SomeFloat and not defined(JS):
     s.addFloat(x)
   elif compiles(s.add(x)):
     s.add(x)
```

But then you might need to update the `addInt` and `addFloat` to be generics?

**Comment by Araq** [comment]
IMO it's not worth fixing, once the deprecated add procs are gone the deprecation warning will go away too.

**Comment by kaushalmodi** [comment]
I am not understanding .. isn't the issue that `addInt` is being used only for `int` type and not for any other `SomeInteger` types? (and the same for float/SomeFloat)

**Comment by ghost** [comment]
Can be closed, doesn't happen on 1.0.4:
https://play.nim-lang.org/#ix=28ag

**Comment by kaushalmodi** [comment]
You are correct. Thanks for following up on this.
